yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMasterArguments.scala

```diff
@@ -92,6 +97,7 @@ class ApplicationMasterArguments(val args: Array[String]) {
       |  --jar JAR_PATH       Path to your application's JAR file
       |  --class CLASS_NAME   Name of your application's main class
       |  --primary-py-file    A main Python file
+      |  --primary-r-file     A main R file
```

Comment:
IIUC if the user wants to run R then s/he shouldn't have python files. Maybe we should validate only one of `primaryRFile` and `primaryPyFile` can be set at a given time.